### ISSUE TITLE
[Typescript] - Error when setting agent configuration

### ISSUE THREAD

**Issue by tjw0051** [amazon-connect/amazon-connect-streams#420]
It looks like there might be an issue setting/getting the agent configuration. If you do:
```typescript
const config = agent.getConfiguration();
agent.setConfiguration(config);
```
I get the error:
```typescript
Uncaught (in promise) ReferenceError: AgentErrorStates is not defined
    at Agent.setConfiguration (connect-streams.js?550c:24094)
    at setDeskphone (actions.ts?2313:110)
    at eval (actions.ts?375c:30)
    at new Promise (<anonymous>)
    at setAudioSettings (actions.ts?375c:14)
    at eval (actions.ts?375c:11)
    at new Promise (<anonymous>)
    at loadAudioSettings (actions.ts?375c:9)
    at Object.eval [as __] (index.tsx?ee8a:114)
    at j (index.js?a178:368)
```

The error response from `amazon-connect-streams` isn't working, I think this line should be `connect.AgentErrorStates`: https://github.com/amazon-connect/amazon-connect-streams/blob/master/src/api.js#L464

But the reason the error gets thrown at all is because the response from `agent.getConfiguration()` doesn't include locale:
```typescript
configuration.agentPreferences.locale = undefined
```

### Workaround
To avoid the error i'm using:
```typescript
const config = agent.getConfiguration();
  const newConfig = {
    ...config,
    agentPreferences: {
      ...config.agentPreferences,
      locale: 'en_US'
    }
  };
  agent.setConfiguration(newConfig)
```

**Comment by cbgandhi** [comment]
Hi @tjw0051, hope you are well.

The workaround didn't work for me, here is my code snippet:

function switchToPhone() {
    console.log("In Swtich Phone");
    var config = myagent.getConfiguration();
    config.extension = "+18005550100";
    config.softphoneEnabled = false;
    config.agentPreferences.locale= 'en';

    myagent.setConfiguration(config, {
        success: function () {
            console.log("Config changed");
        },

        failure: function (err) {
            console.log("Error " + JSON.stringify(err));
        }

    });
}

See if you can help.

Thanks in advance.

**Comment by martin-greenberg** [comment]
I think you're correct that the line in api.js should use `connect.AgentErrorStates.INVALID_LOCALE`. We'll fix this and update this issue when the change is made.

However, per [the documentation](https://github.com/amazon-connect/amazon-connect-streams/blob/master/Documentation.md#agentsetconfiguration) for agent.setConfiguration(), "only the extension and softphoneEnabled values can be updated." We don't currently support setting other fields using this function, and for getConfiguration we only support the documented fields, currently: agentStates, dialableCountries, extension, name, permissions, routingProfile, softphoneAutoAccept, softphoneEnabled, and username.

**Comment by martin-greenberg** [comment]
Closing this issue as the bug is fixed in version 1.6.9.
